### ISSUE TITLE
Print head stays on top of model when print finishes (P1P)

### ISSUE THREAD

**Issue by Hologos** [bambulab/BambuStudio#1242]
**Describe the bug**
The print head stays on top of the model for few seconds when the print finishes.

I think it's caused by the `M400 ; wait for buffer to clear` gcode in Machine end G-code.
```
M400 ; wait for buffer to clear
G92 E0 ; zero the extruder
G1 E-0.8 F1800; retract
G1 Z{max_layer_z + 0.5} F900 ; lower z a little
G1 X65 Y245 F12000 ; move to safe posG1 Y265 F3000
```

According to [Marlin documentation](https://marlinfw.org/docs/gcode/M400.html), this command causes G-code processing to pause and wait in a loop until all moves in the planner are completed. It moves the head to bin position after this command. So maybe moving M400 after those move commands can help.

**3mf File for This Bug**
Nope

**To Reproduce**
1. Start a print.
2. Wait for end.
3. Watch the print head.

**Expected behavior**
It moves the head from the model.

**Screenshots**
Not needed.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS, Windows]

**Comment by SaltWei** [comment]
Printer team will update end-gcode to fix this issue. Thank you very much!

**Comment by eddydas** [comment]
Eagerly looking forward to have this fixed!

**Comment by sodachen22** [comment]
Thanks for the feedback. Yes, it is a bug, and will be fixed.

**Comment by Hologos** [comment]
@sodachen22 This seems to be fixed with latest version v1.7.2.51.

**Comment by Hologos** [comment]
@SaltWei Hello, this can be closed as fixed.
